tiledb/sm/c_api/tiledb.cc

```diff
@@ -2617,6 +2629,119 @@ int32_t tiledb_array_schema_has_attribute(
   return TILEDB_OK;
 }
 
+/* ********************************* */
+/*            SCHEMA EVOLUTION       */
+/* ********************************* */
+
+int32_t tiledb_schema_evolution_alloc(
+    tiledb_ctx_t* ctx, tiledb_schema_evolution_t** schema_evolution) {
+  // Sanity check
+  if (sanity_check(ctx) == TILEDB_ERR)
+    return TILEDB_ERR;
+
+  // Create schema evolution struct
+  *schema_evolution = new (std::nothrow) tiledb_schema_evolution_t;
+  if (*schema_evolution == nullptr) {
+    auto st =
+        Status::Error("Failed to allocate TileDB schema evolution object");
+    LOG_STATUS(st);
+    save_error(ctx, st);
+    return TILEDB_OOM;
+  }
+
+  // Create a new SchemaEvolution object
+  (*schema_evolution)->schema_evolution_ =
+      new (std::nothrow) tiledb::sm::SchemaEvolution();
+  if ((*schema_evolution)->schema_evolution_ == nullptr) {
+    delete *schema_evolution;
+    *schema_evolution = nullptr;
+    auto st =
+        Status::Error("Failed to allocate TileDB schema evolution object");
+    LOG_STATUS(st);
+    save_error(ctx, st);
+    return TILEDB_OOM;
+  }
+
+  // Success
+  return TILEDB_OK;
+}
+
+int32_t tiledb_schema_evolution_add_attribute(
+    tiledb_ctx_t* ctx,
+    tiledb_schema_evolution_t* schema_evolution,
+    tiledb_attribute_t* attr) {
+  if (sanity_check(ctx) == TILEDB_ERR ||
+      sanity_check(ctx, schema_evolution) == TILEDB_ERR ||
+      sanity_check(ctx, attr) == TILEDB_ERR)
+    return TILEDB_ERR;
+
+  if (SAVE_ERROR_CATCH(
+          ctx, schema_evolution->schema_evolution_->add_attribute(attr->attr_)))
+    return TILEDB_ERR;
+  return TILEDB_OK;
+
+  // Success
+  return TILEDB_OK;
+}
+
+int32_t tiledb_schema_evolution_drop_attribute(
+    tiledb_ctx_t* ctx,
+    tiledb_schema_evolution_t* schema_evolution,
+    const char* attribute_name) {
+  if (sanity_check(ctx) == TILEDB_ERR ||
+      sanity_check(ctx, schema_evolution) == TILEDB_ERR)
+    return TILEDB_ERR;
+
+  if (SAVE_ERROR_CATCH(
+          ctx,
+          schema_evolution->schema_evolution_->drop_attribute(attribute_name)))
+    return TILEDB_ERR;
+  return TILEDB_OK;
+  // Success
+  return TILEDB_OK;
+}
+
+int32_t tiledb_array_evolve(
```

Comment:
This needs to be moved to the `ARRAY` API functions in this file (same for the header).